packages/flutter/lib/src/widgets/route_notification_messages.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+import 'navigator.dart';
+
+/// Messages for route change notifications.
+class RouteNotificationMessages {
+  RouteNotificationMessages._();
+
+  /// When the engine is Web notify the platform for a route change.
+  static void maybeNotifyRouteChange(String methodName, Route<dynamic> route, Route<dynamic> previousRoute) {
+    if(kIsWeb) {
```

Comment:
Why is this hidden behind the kIsWeb flag? We should just always send it and on platforms who don't care the engine just ignores it.

Reply:
Also: (nit) there should be a space after the if.